src/mappings/entities/mapToken.ts

```diff
@@ -0,0 +1,165 @@
+import BN from 'bn.js';
+import { SubstrateExtrinsic } from '@subql/types';
+import { ModuleIdEnum, CallIdEnum } from './common';
+import { Token } from '../../types';
+
+const chainNumberMultiplier = new BN(1000000);
+
+export async function mapToken(
+  blockId: number,
+  callId: CallIdEnum,
+  moduleId: ModuleIdEnum,
+  params: Record<string, any>,
+  extrinsic: SubstrateExtrinsic,
+): Promise<void> {
+  if (!extrinsic.success || ![ModuleIdEnum.Asset].includes(moduleId)) {
+    return;
+  }
+
+  if (callId === CallIdEnum.CreateAsset) {
+    const {
+      name,
+      ticker,
+      assetType: type,
+      fundingRound,
+      divisible,
+      disableIu,
+      identifiers,
+    } = params;
+    const ownerDid = extrinsic.events[1].event.data[0].toString();
+    await Token.create({
+      id: ticker,
+      ticker,
+      name,
+      type,
+      fundingRound,
+      isDivisible: divisible,
+      isFrozen: false,
+      isUniquenessRequired: !disableIu,
+      documents: [],
+      identifiers: JSON.stringify(identifiers),
+      ownerDid,
+      fullAgents: [ownerDid],
+      holders: [],
+      totalSupply: '0',
+    }).save();
+  }
+
+  if (callId === CallIdEnum.RenameAsset) {
+    const { ticker, name: newName } = params;
+    const token = await Token.getByTicker(ticker);
+    if (!token) return;
+    await Token.create({
```

Comment:
What about, 
```
token.name = newName
await token.save()
```
?

I don't think we should  reinstantiate an instance we already have.

(applies to each case)

Reply:
Ah cool, didn't know I can do that. Didn't see it docs either. But it looked liked every time you do `.create().save()` it basically upserts.